rest_framework/schemas/openapi.py

```diff
@@ -442,11 +443,17 @@ def _map_field(self, field):
                 content['format'] = field.protocol
             return content
 
-        # DecimalField has multipleOf based on decimal_places
         if isinstance(field, serializers.DecimalField):
-            content = {
-                'type': 'number'
-            }
+            if getattr(field, 'coerce_to_string', api_settings.COERCE_DECIMAL_TO_STRING):
+                content = {
+                    'type': 'string',
+                    'format': 'decimal',
```

Comment:
I went through https://github.com/OAI/OpenAPI-Specification/issues/845#issuecomment-378139730.

I found that the `decimal` format is supported for `number` datatype as well.

The rest of the code looks good to me!

Reply:
I constantly struggle to find answers spelled out clearly and canonically. Do we want to add this? 
https://github.com/OAI/OpenAPI-Specification/issues/845 is still WIP, is it actually part of the spec yet?

Inclined to take this as it is now, and add more later...